### PR TITLE
Increase code coverage of MindtPy

### PR DESCRIPTION
**Pull request by ZedongPeng**
## Summary/Motivation:
This pull request includes the changes to increase code coverage of MindtPy.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by jsiirola — CHANGES_REQUESTED** [review]
This looks pretty good.  Two big topics that would be good to address:
- removing unused local variables
- making sure that the tests actually confirm that the code is actually doing the desired behavior 
   - In addition to notes on specific tests below, the solver regression tests are only checking the objective value.  It would improve their robustness (and usefulness) if they also verified the state variables.

**Comment by jsiirola** [comment]
@ZedongPeng, quick question on this: is there a timeline to add asserts to the two tests in `unit_test.py`?  Apart from that, I think this PR is good to merge...

**Comment by ZedongPeng** [comment]
> @ZedongPeng, quick question on this: is there a timeline to add asserts to the two tests in `unit_test.py`? Apart from that, I think this PR is good to merge...

@jsiirola Sorry for the late progress. I have just added assert in the two tests. Please let me know if anything else needs to be changed before merging the PR.

**Review by jsiirola — APPROVED** [review]

File: pyomo/contrib/mindtpy/tests/test_mindtpy_ECP.py
Comment (mrmundt) [reply pyomo/contrib/mindtpy/tests/test_mindtpy_ECP.py:78]: (Second review) Could you please document why these commented-out tests should not be deleted?

**Review by mrmundt — APPROVED** [review]

File: pyomo/contrib/mindtpy/tests/test_mindtpy_ECP.py
Comment (ZedongPeng) [reply pyomo/contrib/mindtpy/tests/test_mindtpy_ECP.py]: Sorry. I forgot to remove these commented tests. I have removed them now.

**Pull request merged**
